src/test/scala/sangria/execution/QueryReducerSpec.scala

```diff
@@ -789,4 +790,71 @@ class QueryReducerSpec extends WordSpec with Matchers with FutureResultSupport {
       error.cause should be (IntrospectionNotAllowedError)
     }
   }
-}
\ No newline at end of file
+  case class Context(seen: Map[String, String])
+
+    "QueryReducerExecutor" should {
+
+      val query = graphql"""
+         query myQuery($$include: Boolean!, $$skip: Boolean!) {
+           nest {
+             skip: a @skip(if: $$skip)
+             include: a @include(if: $$include)
+             a
+           }
+         }
+       """
+      
+      val reducer = new QueryReducer[Info, Info] {
+        override type Acc = Seq[Int]
+
+        override def initial: Acc = Seq.empty
+
+        override def reduceAlternatives(alternatives: Seq[Acc]): Acc = ???
+
+        override def reduceField[Val](
+          fieldAcc: Acc,
+          childrenAcc: Acc,
+          path: ExecutionPath,
+          ctx: Info,
+          astFields: Vector[ast.Field],
+          parentType: ObjectType[Info, Val],
+          field: Field[Info, Val],
+          argumentValuesFn: ArgumentValuesFn
+        ): Acc = {
+          val args = argumentValuesFn(path, field.arguments, astFields.head.arguments)
+          assert(args == Success(Args.empty))
+          fieldAcc ++ childrenAcc :+ 1
+        }
+
+        override def reduceScalar[T](
+          path: ExecutionPath,
+          ctx: Info,
+          tpe: ScalarType[T]
+        ): Acc = initial
+
+        override def reduceEnum[T](
+          path: ExecutionPath,
+          ctx: Info,
+          tpe: EnumType[T]
+        ): Acc = ???
+
+        override def reduceCtx(
+          acc: Acc,
+          ctx: Info
+        ): ReduceAction[Info, Info] = ctx.copy(ctx.nums ++ acc)
+      }
+
+      "work with variables absent" in {
+        val res =
+          QueryReducerExecutor.reduceQueryWithoutVariables(
+            schema,
+            query,
+            Info(Seq(100)),
+            (),
+            reducer :: Nil
+          )
+          .await
+        assert(res._1.nums == Seq(100, 1, 1, 1, 1), "expected 4 nodes to be visited")
+      }
+    }
```

Comment:
Maybe you can also add extra tests with some existing query reduces and without variables? (`MeasureComplexity` in particular since it would be the main use-case, as far as I understand) You probably will need to define different schema just for this test because complexity function would have a different implementation

Reply:
I added a test with MeasureComplexity, let me know if you had more in mind